src/Client.js

```diff
@@ -815,21 +818,71 @@ class Client extends EventEmitter {
             );
         }
 
+
+        const isBigFile = internalOptions.attachment?.data?.length > (1024 * 1024 * 79);
+
+        if (isBigFile) {
+			let startDivision = 2
+            let middle = internalOptions.attachment.data.length / startDivision;
+            let currentIndex = 0;
+			let passes = 0;
+            
+			while (middle > (1024 * 1024 * 50)){
+				startDivision += 1
+				middle = Math.floor(internalOptions.attachment.data.length / startDivision);
+			}
+            
+			while(currentIndex < internalOptions.attachment.data.length){
+				let chunkPiece = middle
+				if(currentIndex + middle > internalOptions.attachment.data.length){
+					chunkPiece = internalOptions.attachment.data.length - currentIndex
+				}
+				await this.pupPage.evaluate(async (chatId, chunk,passes) => {
+					if (chunk) {
+						window.Store[`mediaChunk_${chatId}_${passes}`] = chunk;
+					}
+				}, chatId, internalOptions.attachment.data.substring(currentIndex, currentIndex+chunkPiece),passes);
+				currentIndex += chunkPiece
+				passes += 1
+			}
+            
+            internalOptions.attachment = new MessageMedia(internalOptions.attachment.mimetype,"",internalOptions.attachment.filename,internalOptions.attachment.filesize);
+            
+			await this.pupPage.evaluate(async (chatId,passes) => {
+					window.Store[`mediaChunk_${chatId}_passes`] = passes;	
+			}, chatId,passes);
+        }
+
```

Comment:
I tried to improve is a little:
```js
        const isBigFile = internalOptions.attachment?.data?.length > (1024 * 1024 * 79);

        if (isBigFile) {
			let startDivision = 2
            let middle = internalOptions.attachment.data.length / startDivision;
            let currentIndex = 0;

            
			while (middle > (1024 * 1024 * 50)){
				startDivision += 1
				middle = Math.floor(internalOptions.attachment.data.length / startDivision);
			}
            
            const randomId = Util.generateHash(32);
            
			while(currentIndex < internalOptions.attachment.data.length){
				let chunkPiece = middle
				if(currentIndex + middle > internalOptions.attachment.data.length){
					chunkPiece = internalOptions.attachment.data.length - currentIndex
				}
				await this.pupPage.evaluate(async (chatId, chunk, randomId) => {
					if (chunk && window[`mediaChunk_${randomId}`]) {
						window[`mediaChunk_${randomId}`] += chunk;
					}
                    else {
                        window[`mediaChunk_${randomId}`] = chunk;
                    }
				}, chatId, internalOptions.attachment.data.substring(currentIndex, currentIndex+chunkPiece), randomId);
				currentIndex += chunkPiece

			}
            
            internalOptions.attachment = new MessageMedia(internalOptions.attachment.mimetype,`mediaChunk_${randomId}`, internalOptions.attachment.filename,internalOptions.attachment.filesize);
        }
```

Reply:
Fun history
I tried this initially and still crashed..., and forgot about it (i dont have much knowledge on js and puppeteer)
Later i figured that my problem was actually the attachment data not getting overwritten if setting it directly (so i changed to instantiante the whole object into a new MessageMedia)

Later once everything worked, i completely forgot that my previous logic was correct and i had to change it back

Good catch, i will push your changes and modify the PR

Thanks!